src/components/Header.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useContext } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+import { searchResultsContext } from '../contexts/contexts';
+
+export default function Header({ handleLoginClick, handleSignupClick }) {
+  const navigate = useNavigate();
+  const { setSearchResults } = useContext(searchResultsContext);
+  const [query, setQuery] = useState(''); // User input
+
+  const setBooks = (books) => {
+    setSearchResults(books);
+  };
+
+  const getLargerImageUrl = (thumbnailUrl) => {
+    if (thumbnailUrl) {
+      return thumbnailUrl
+        .replace('&zoom=1', '&zoom=2')
+        .replace('http', 'https');
+    } else {
+      return 'https://via.placeholder.com/128x192.png?text=No+Image';
+    }
+  };
+
+  // Function to handle the API request
+  const fetchBooks = async () => {
+    if (!query.trim()) return; // Avoid empty searches
+
+    try {
+      const response = await fetch(
+        `https://www.googleapis.com/books/v1/volumes?q=${encodeURIComponent(
+          query
+        )}&maxResults=40`

```

Comment:
All `api` functions should be defined in `utils`, and only then used in the project. You need to split the code